mmengine/hooks/freeze_hook.py

```diff
@@ -0,0 +1,121 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from typing import Optional, Sequence, Union
+
+from mmengine.logging import print_log
+from mmengine.model import is_model_wrapper
+from mmengine.registry import HOOKS
+from .hook import Hook
+
+
+@HOOKS.register_module()
+class FreezeHook(Hook):
+    """FreezeHook is used to freeze or unfreeze network layers when training to
+    a specified epoch.
+
+    Args:
+        freeze_epoch (int): The epoch number to start freezing layers.
+        freeze_layers (tuple[str]): Model layers containing the keyword in
+            freeze_layers will freeze the gradient.
+        unfreeze_epoch (int): The epoch number to start unfreezing layers.
```

Comment:
Please refer to https://github.com/open-mmlab/mmengine/blob/6c5eebb823e3c9381d63fd0cd1873ed1bd9ee9de/mmengine/hooks/ema_hook.py#L32 updating `freeze_iter` and `unfreeze_iter`